### ISSUE TITLE
Compile failure with GHC 7.8

### ISSUE THREAD

**Issue by hvr** [passy/giphy-api#12]
The Hackage matrix builder detected the following compile failure:

```
Configuring component lib from giphy-api-0.5.1.0
Preprocessing library giphy-api-0.5.1.0...
[1 of 2] Compiling Paths_giphy_api  ( /tmp/matrix-worker/1480361044/dist-newstyle/build/x86_64-linux/ghc-7.8.4/giphy-api-0.5.1.0/build/autogen/Paths_giphy_api.hs, /tmp/matrix-worker/1480361044/dist-newstyle/build/x86_64-linux/ghc-7.8.4/giphy-api-0.5.1.0/build/Paths_giphy_api.o )
[2 of 2] Compiling Web.Giphy        ( src/Web/Giphy.hs, /tmp/matrix-worker/1480361044/dist-newstyle/build/x86_64-linux/ghc-7.8.4/giphy-api-0.5.1.0/build/Web/Giphy.o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package array-0.5.0.0 ... linking ... done.
Loading package deepseq-1.3.0.2 ... linking ... done.
Loading package bytestring-0.10.4.0 ... linking ... done.
Loading package containers-0.5.5.1 ... linking ... done.
Loading package fail-4.9.0.0 ... linking ... done.
Loading package binary-0.7.1.0 ... linking ... done.
Loading package text-1.2.2.1 ... linking ... done.
Loading package hashable-1.2.4.0 ... linking ... done.
Loading package transformers-0.5.2.0 ... linking ... done.
Loading package primitive-0.6.1.0 ... linking ... done.
Loading package vector-0.11.0.0 ... linking ... done.
Loading package scientific-0.3.4.9 ... linking ... done.
Loading package pretty-1.1.1.1 ... linking ... done.
Loading package template-haskell ... linking ... done.
Loading package nats-1.1.1 ... linking ... done.
Loading package transformers-compat-0.5.1.4 ... linking ... done.
Loading package tagged-0.8.5 ... linking ... done.
Loading package unordered-containers-0.2.7.1 ... linking ... done.
Loading package semigroups-0.18.2 ... linking ... done.
Loading package attoparsec-0.13.1.0 ... linking ... done.
Loading package old-locale-1.0.0.6 ... linking ... done.
Loading package time-1.4.2 ... linking ... done.
Loading package unix-2.7.0.1 ... linking ... done.
Loading package base-compat-0.9.1 ... linking ... done.
Loading package dlist-0.8.0.2 ... linking ... done.
Loading package time-locale-compat-0.1.1.3 ... linking ... done.
Loading package aeson-1.0.2.1 ... linking ... done.
Loading package base64-bytestring-1.0.0.1 ... linking ... done.
Loading package mtl-2.2.1 ... linking ... done.
Loading package stm-2.4.4.1 ... linking ... done.
Loading package exceptions-0.8.3 ... linking ... done.
Loading package blaze-builder-0.4.0.2 ... linking ... done.
Loading package uri-bytestring-0.2.2.1 ... linking ... done.
Loading package random-1.1 ... linking ... done.
Loading package uuid-types-1.0.3 ... linking ... done.
Loading package http-api-data-0.3.3 ... linking ... done.
Loading package case-insensitive-1.2.0.7 ... linking ... done.
Loading package data-default-class-0.1.2.0 ... linking ... done.
Loading package cookie-0.4.2.1 ... linking ... done.
Loading package filepath-1.3.0.2 ... linking ... done.
Loading package http-types-0.9.1 ... linking ... done.
Loading package mime-types-0.1.0.7 ... linking ... done.
Loading package network-2.6.3.1 ... linking ... done.
Loading package parsec-3.1.11 ... linking ... done.
Loading package network-uri-2.6.1.0 ... linking ... done.
Loading package async-2.1.1 ... linking ... done.
Loading package directory-1.2.1.0 ... linking ... done.
Loading package process-1.2.0.0 ... linking ... done.
Loading package zlib-0.6.1.2 ... linking ... done.
Loading package streaming-commons-0.1.16 ... linking ... done.
Loading package http-client-0.5.4 ... linking ... done.
Loading package byteable-0.1.1 ... linking ... done.
Loading package cereal-0.5.4.0 ... linking ... done.
Loading package socks-0.5.5 ... linking ... done.
Loading package hourglass-0.2.10 ... linking ... done.
Loading package memory-0.13 ... linking ... done.
Loading package asn1-types-0.3.2 ... linking ... done.
Loading package asn1-encoding-0.9.4 ... linking ... done.
Loading package cryptonite-0.20 ... linking ... done.
Loading package asn1-parse-0.9.4 ... linking ... done.
Loading package pem-0.2.2 ... linking ... done.
Loading package x509-1.6.4 ... linking ... done.
Loading package x509-store-1.6.2 ... linking ... done.
Loading package x509-validation-1.6.5 ... linking ... done.
Loading package tls-1.3.8 ... linking ... done.
Loading package x509-system-1.6.4 ... linking ... done.
Loading package connection-0.2.6 ... linking ... done.
Loading package http-client-tls-0.3.3 ... linking ... done.
Loading package http-media-0.6.4 ... linking ... done.
Loading package safe-0.3.10 ... linking ... done.
Loading package mmorph-1.0.9 ... linking ... done.
Loading package utf8-string-1.0.1.1 ... linking ... done.
Loading package string-conversions-0.4 ... linking ... done.
Loading package vault-0.3.0.6 ... linking ... done.
Loading package servant-0.9.1.1 ... linking ... done.
Loading package servant-client-0.9.1.1 ... linking ... done.
Loading package microlens-0.4.7.0 ... linking ... done.
Loading package microlens-th-0.4.1.0 ... linking ... done.

src/Web/Giphy.hs:170:12:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:170:25:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:171:12:
    Not in scope: ‘<*>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:171:25:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:172:12:
    Not in scope: ‘<*>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:172:25:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:173:12:
    Not in scope: ‘<*>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:173:25:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:174:12:
    Not in scope: ‘<*>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:174:25:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:175:12:
    Not in scope: ‘<*>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:175:25:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:176:12:
    Not in scope: ‘<*>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:176:25:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:177:12:
    Not in scope: ‘<*>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)

src/Web/Giphy.hs:177:25:
    Not in scope: ‘<$>’
    Perhaps you meant ‘<>’ (imported from Data.Monoid)
```

**Comment by passy** [comment]
Yeah, I don't support GHC 7.8 for the newer releases anymore. But if it's only the Data.Monoid import, I'm happy to add that.

**Comment by hvr** [comment]
@passy if you don't want to support GHC 7.8 / base-4.7 that's totally reasonable. All you have to do is add a lower bound on base, i.e. `base >= 4.8`.

You can simply fixup the meta-data via 

https://hackage.haskell.org/package/giphy-api-0.5.1.0/giphy-api.cabal/edit

**Comment by passy** [comment]
@hvr Thanks for reporting this and the explanations. I've published `0.5.2.0` to hackage; I hope this will work for you!
